clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6272,6 +6272,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (getToolChain().getTriple().getSubArch() ==
         llvm::Triple::SPIRSubArch_fpga)
       CmdArgs.push_back("-D__ENABLE_USM_ADDR_SPACE__");
+
+#if defined(_WIN32)
+    // SYCL library is guaranteed to work correctly only with dynamic runtime.
+    if (!D.IsCLMode()) {
```

Comment:
```suggestion
    if (!D.IsCLMode() && C.getDefaultToolChain().getTriple().isWindowsMSVCEnvironment();) {
```
It would seem to me that using the toolchain environment setting over `_WIN32` would be preferred.